ui_test/src/tests.rs

```diff
@@ -29,25 +32,277 @@ fn main() {
+            messages_from_unknown_file_or_line: vec![],
+            messages: vec![
+                vec![],
+                vec![],
+                vec![],
+                vec![],
+                vec![],
+                vec![
+                    Message {
+                        message: "Undefined Behavior: type validation failed: encountered a dangling reference (address 0x10 is unallocated)".to_string(),
+                        level: Level::Error,
+                    },
+                    Message {
+                        message: "kaboom".to_string(),
+                        level: Level::Warn,
+                    },
+                    Message {
+                        message: "cake".to_string(),
+                        level: Level::Warn,
+                    },
+                ],
+            ],
+        };
+        let mut errors = vec![];
+        check_annotations(diagnostics, Path::new("moobar"), &mut errors, &config, "", &comments);
+        match &errors[..] {
+            [Error::ErrorsWithoutPattern { line: 5, msgs, .. }] =>
+                match &msgs[..] {
+                    [Message { message, level: Level::Warn }] if message == "kaboom" => {}
+                    _ => panic!("{:#?}", msgs),
+                },
+            _ => panic!("{:#?}", errors),
+        }
+    }
+}
+
+#[test]
+fn missing_implicit_warn_pattern() {
+    let s = r"
+use std::mem;
+
+fn main() {
+    let _x: &i32 = unsafe { mem::transmute(16usize) }; //~ ERROR encountered a dangling reference (address 0x10 is unallocated)
+    //~^ cake
+}
+    ";
+    let comments = Comments::parse(Path::new("<dummy>"), s);
+    let config = config();
+    let unnormalized_stderr =
+        r"this isn't checked anymore anyway and only used for test failure messages";
+    {
+        let diagnostics = Diagnostics {
+            rendered: unnormalized_stderr.to_string(),
+            messages_from_unknown_file_or_line: vec![],
+            messages: vec![
+                vec![],
+                vec![],
+                vec![],
+                vec![],
+                vec![],
+                vec![
+                    Message {
+                        message: "Undefined Behavior: type validation failed: encountered a dangling reference (address 0x10 is unallocated)".to_string(),
+                        level: Level::Error,
+                    },
+                    Message {
+                        message: "kaboom".to_string(),
+                        level: Level::Warn,
+                    },
+                    Message {
+                        message: "cake".to_string(),
+                        level: Level::Warn,
+                    },
+                ],
+            ],
+        };
+        let mut errors = vec![];
+        check_annotations(diagnostics, Path::new("moobar"), &mut errors, &config, "", &comments);
+        match &errors[..] {
+            [Error::ErrorsWithoutPattern { line: 5, msgs, .. }] =>
```

Comment:
Nice test! As indicated by my previous comment, however, I think this should *not* have any error.